internal/pkg/errors/errors.go

```diff
@@ -147,11 +147,41 @@ To enable it, run:
 
 	IAAS_SERVER_MISSING_IMAGE_OR_VOLUME_FLAGS = `Either "image-id" or "boot-volume-source-type" and "boot-volume-source-id" flags must be provided.`
 
+	IAAS_SECURITY_GROUP_RULE_PROTOCOL_PORT_RANGE_CONFLICT = `"port-range-min" and "port-range-max" attributes can't be provided if "protocol-name" is set to "icmp" or "ipv6-icmp"`
+
+	IAAS_SECURITY_GROUP_RULE_PROTOCOL_PARAMETERS_CONFLICT = `"icmp-parameter-code" and "icmp-parameter-type" attributes can't be provided if "protocol-name" is not "icmp" or "ipv6-icmp"`
```

Comment:
These two already seem a bit validation that could be done in the API. Or is the error not clear?

Reply:
The error messages from API response are very clear. Very similar to our error message. Also same for the `IAAS_SECURITY_GROUP_RULE_PROTOCOL_MISSING_FOR_PORT_RANGE`

I can remove the error handling from our side for the following error messages

1. IAAS_SECURITY_GROUP_RULE_PROTOCOL_PORT_RANGE_CONFLICT
2. IAAS_SECURITY_GROUP_RULE_PROTOCOL_PARAMETERS_CONFLICT
3. IAAS_SECURITY_GROUP_RULE_PROTOCOL_MISSING_FOR_PORT_RANGE